### ISSUE TITLE
Standardize library path

### ISSUE THREAD

**Issue by maxux** [threefoldtech/js-sdk#1182]
Configuration files are stored at ~/.config/jumpscale which is good standard.
But then you have `~/sandbox` which is completely wrong.

Jumpscale library code, etc. should be installed at `~/.local/lib` (and `~/.local/bin` for binaries, if any).
This would completely standardize files path and looks more like real unix package :)

**Comment by xmonader** [comment]
I agree with that for sure, let's try to follow https://www.freedesktop.org/software/systemd/man/file-hierarchy.html as much as we can.

Here's the current [implementation](https://github.com/threefoldtech/js-ng/blob/aa8600dacbd62f1e43f90f0f0d9a871d3289d670/jumpscale/core/dirs/dirs.py#L45) of j.dirs

```python
class Dirs:

    HOMEDIR = os.path.expanduser("~")  # TODO: check homedir defined in sal.fs
    BASEDIR = os.path.join(HOMEDIR, "sandbox")
    BINDIR = os.path.join(BASEDIR, "bin")
    CFGDIR = os.path.join(BASEDIR, "cfg")  # TODO: check conflict with core.config_root..
    CODEDIR = os.path.join(BASEDIR, "code")
    VARDIR = os.path.join(BASEDIR, "var")
    LOGDIR = os.path.join(VARDIR, "log")
    JSCFGDIR = os.path.join(HOMEDIR, ".config/jumpscale")
    TEMPLATEDIR = os.path.join(VARDIR, "templates")
    TMPDIR = "/tmp/jumpscale"
```
I think when revisiting we need to consider the following 

- we don't need `TEMPLATEDIR` anymore I think
- better we introduce CACHEDIR (for downloads we do instead of `~/sandbox/var/downloaded_packages/`) or should be `~/.jumpscale/var/downloaded_packages`
- `~/.local/lib/jumpscale` ideally would have been used if we have some vendor data (right now that's completely managed by virtualenv, and when installing using pip it will use the /usr/lib or ~/.local/lib to install the deps
- `CODEDIR` by convention is `~/sandbox/code` which we depend on in the git sal for cloning repos
- apps configurations (e.g nginx, redis, .. etc) I think they should go under some directory maybe `~/.config/jumpscale/etc`, but here it raises a problem what if we want to discard that dir for wrong configurations or regenerate everytime e.g in case of something dynamic like nginx? this directory `~/.config/jumpscale` is backed up by convention and if something wrong you won't be able to have a successful restore
- Some apps e.g zoom, komodo, thunderbird some their config/logs in `$HOME/.$appname`, so maybe `~/.jumpscale/{etc,cfg,var,logs,cache}` can be used for logs or then maybe it's okay to have sandbox dir`~/sandbox/{etc,cfg,var,logs,cache}`, and transient configs, logs and caching?
```
➜  ~ ls ~/.zoom 
data  im  logs  reports  screenCapture  shareItems
➜  ~ ls .thunderbird 
'Crash Reports'   i676rnyr.default  'Pending Pings'   profiles.ini
➜  ~ ls ~/.kiki 
kikicfg.py
➜  ~ ls ~/.komodoide 
12.0
➜  ~ ls ~/.komodoide/12.0/
autosave            codeintel3.log  history.sqlite.bak    prefs.xmlc    samples          tools
breakpoints.pickle  colors.less     interfaceChrome.less  pystderr.log  schemes          userChrome.less
classic.less        doc-state.xmlc  mutex.lock            pystdout.log  startup-env.tmp  view-state.xmlc
codeintel3.db       history.sqlite  prefs.xml             running.lock  toolbox.sqlite   XRE

```

if we are going to create `~/.jumpscale` directory, wouldn't that be confused with configs of `~/.config/jumpscale` or we merge both in `~/.jumpscale`. What do you think @maxux?
